### PR TITLE
[3.10] Add Joomla 3.10 to Joomla 4.0 nightly update tests

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

Add Joomla 3.10 to Joomla 4.0 nightly update tests

### Testing Instructions

The tests should run an update from 3.10 to the latest 4 nightly

### Expected result

The tests passes (hopefully)

### Actual result

there is no test to pass

### CODE REVIEW

File: tests/codeception/acceptance/updateto4/Updateto4Cest.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	 *
+	 * @param   AcceptanceTester  $I  The AcceptanceTester Object
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 *
+	 * @return  void
+	 */
+	public function updateToJoomla4(\AcceptanceTester $I)
+	{
+		$I->comment('I am going to create a user');
```
Comment (HLeithner) [inline tests/codeception/acceptance/updateto4/Updateto4Cest.php]: You create a user here?

Comment (zero-24) [reply tests/codeception/acceptance/updateto4/Updateto4Cest.php]:
```suggestion
		$I->comment('I am going to test 3.10 to 4.0 nightly updates');
```

Comment (zero-24) [reply tests/codeception/acceptance/updateto4/Updateto4Cest.php]: should be fixed now. Thanks

File: RoboFile.php [hunk 2]
```diff
@@ -284,16 +281,60 @@ public function runTests($opts = ['use-htaccess' => false, 'env' => 'desktop'])
...
+		// Make sure to run the build command to generate AcceptanceTester
+		if ($this->isWindows())
+		{
+			$this->_exec('php ' . $this->getWindowsPath($this->testsPath . 'vendor/codeception/codeception/codecept') . ' build');
+			$pathToCodeception = $this->getWindowsPath('php ' . $this->testsPath . 'vendor/codeception/codeception/codecept');
+		}
+		else
+		{
+			$this->_exec('php ' . $this->testsPath . 'vendor/bin/codecept build');
+
```
Comment (brianteeman) [inline RoboFile.php]:
```suggestion
```

File: RoboFile.php [hunk 3]
```diff
@@ -284,16 +281,60 @@ public function runTests($opts = ['use-htaccess' => false, 'env' => 'desktop'])
...
+			->arg($this->testsPath . '/acceptance/administrator/components/com_users')
+			->run()
+			->stopOnFail();
+	}
+
+	/**
+	 * Executes the Joomla 3.10 to 4.x test
+	 *
+	 * @param   array $opts   Array of configuration options:
+	 *                        - 'use-htaccess': renames and enable embedded Joomla .htaccess file
```
Comment (brianteeman) [inline RoboFile.php]:
```suggestion
	 *                        - 'use-htaccess': rename and enable embedded Joomla .htaccess file
```

**Comment by zero-24** [comment]
Finally we made it:
image

Drone is happy now :D Many thanks to @yvesh for your support. cc @Hackwar

**Comment by wilsonge** [comment]
Drone succeeding is a bad thing tho. Right now the nightlies are failing to upgrade because of https://github.com/joomla/joomla-cms/issues/28432 - so I'd hope drone is failing right now

**Comment by zero-24** [comment]
> Drone succeeding is a bad thing tho.

Yes this is expected as mention above when the issue has been fixed and merged we can finally fix this here too. It took just a tone of commits to get the system tests running again and we are now at the state that we can get forward when the the issue mention are fixed.

**Comment by wilsonge** [comment]
It’s been merged so can we uncomment so we have a before and after tonight’s nightly build?

**Comment by zero-24** [comment]
I can take a look into the test tomorrow with that patch merged and the new nighly build

**Pull request closed without merging**